### PR TITLE
Exercise-2: otp-field

### PR DESCRIPTION
**Pull request by bhoomika08**

### CODE REVIEW

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if (key === 'Backspace' && index !== 0) {
+      otp[index - 1].ref.current.focus();
+    }
+  }
+
+  focusNext(index, value) {
+    const {otp} = this.state;
+    const {getOtp} = this.props;
+    const isNumericValue = numRegex.test(value);
+    if (isNumericValue) {
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: You can simply remove the nested if-else, like this
if(!isNumericValue) {
  return
}
...

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      if (index < otp.length - 1 && value) {
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: If you have already checked the value with regex, then can we remove `&& value` condition?

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+              style={[
+                styles.otpField,
+                isActive ? styles.activeOtpField : styles.deactiveOtpField,
+              ]}
+              onFocus={() => this.activateInput(idx)}
+              onBlur={() => this.deactivateInput(idx)}
+              maxLength={1}
+              value={otpValue}
+              onChangeText={(val) => this.focusNext(idx, val)}
+              onKeyPress={(event) =>
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: can we work this by listening to a single event?

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js (hunk 1 above)
Comment (bhoomika08) [reply [Exercise-2] Otp-Field/src/components/OtpInputs.js]: Done

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js (hunk 2 above)
Comment (bhoomika08) [reply [Exercise-2] Otp-Field/src/components/OtpInputs.js]: Done

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js (hunk 3 above)
Comment (bhoomika08) [reply [Exercise-2] Otp-Field/src/components/OtpInputs.js]: Removed onKeyPress prop and handled backspace (value == "") in onChangeText.
in onChangeText, we don't get key that is why I used onKeyPress.

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+  render() {
+    const {inputs} = this.props;
+    const {otp} = this.state;
+    return (
+      <View style={styles.otpView}>
+        {inputs.map((input, idx) => {
+          const {value, isActive, ref} = otp[idx];
+          const numOtpVal = Number(value);
+          const otpValue = numOtpVal || numOtpVal == 0 ? numOtpVal : '';
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: === use triple equal to, everywhere

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          return (
+            <TextInput
+              key={`otp-${idx}`}
+              keyboardType="numeric"
+              autoFocus={idx == 0 ? true : false}
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: I think you can simply write `idx === 0` here.

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+    const isNumericValue = numRegex.test(value);
+    if (!isNumericValue) {
+      return;
+    }
+    if (index < otp.length - 1) {
+      otp[index + 1].ref.current.focus();
+    }
+    otp[index]['value'] = value;
+    this.setState({otp});
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: use setOtp here as well

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+    if (index === otp.length - 1) {
+      otp[index].ref.current.blur();
+      const otpValue = otp.map(({value}) => value);
+      getOtp(otpValue.join(''));
+    }
+  }
+
+  activateInput(index) {
+    const {otp} = this.state;
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js:39]: don't make direct changes to the state, instead create a new hash here using JSON deconstruction,

File: [Exercise-2] Otp-Field/src/components/OtpInputs.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+      return;
+    }
+    if (index < otp.length - 1) {
+      otp[index + 1].ref.current.focus();
+    }
+    otp[index]['value'] = value;
+    this.setState({otp});
+
+    if (index === otp.length - 1) {
+      otp[index].ref.current.blur();
```
Comment (sawangupta92) [inline [Exercise-2] Otp-Field/src/components/OtpInputs.js]: take out these three lines 36-38 in a different method

**Pull request merged**